File: backend/src/main/java/com/back/domain/applicant/entity/Applicant.java
package com.back.domain.applicant.entity;

import com.back.domain.adoption.entity.Adoption;
import com.back.domain.care.entity.Care;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

@Entity
@Getter
@Table(name = "applicant")
@EntityListeners(AuditingEntityListener.class)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Applicant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "applicant_id")
    private Long id;

    @Column(name = "applicant_name", nullable = false)
    private String name;

    @Column(name = "applicant_phone", nullable = false)
    private String phone;

    @Column(name = "applicant_email", nullable = false)
    private String email;

    @Column(name = "applicant_address", nullable = false)
    private String address;

    @OneToOne(mappedBy = "applicant")
    private Adoption adoption;

    @OneToOne(mappedBy = "applicant")
    private Care care;

    @Builder
    public Applicant(String name, String phone, String email, String address,
                    Adoption adoption, Care care) {
        this.name = name;
        this.phone = phone;
        this.email = email;
        this.address = address;
        this.adoption = adoption;
        this.care = care;
    }

}

File: backend/src/main/java/com/back/domain/applicant/entity/Applicant.kt
package com.back.domain.applicant.entity

import com.back.domain.adoption.entity.Adoption
import com.back.domain.care.entity.Care
import jakarta.persistence.*
import org.springframework.data.jpa.domain.support.AuditingEntityListener

@Entity
@Table(name = "applicant")
@EntityListeners(AuditingEntityListener::class)
class Applicant private constructor(
    @field:Column(name = "applicant_name", nullable = false)
    var name: String,
    
    @field:Column(name = "applicant_phone", nullable = false)
    var phone: String,
    
    @field:Column(name = "applicant_email", nullable = false)
    var email: String,
    
    @field:Column(name = "applicant_address", nullable = false)
    var address: String,
    
    @field:OneToOne(mappedBy = "applicant")
    var adoption: Adoption?,
    
    @field:OneToOne(mappedBy = "applicant")
    var care: Care?
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "applicant_id")
    var id: Long? = null
        private set

    companion object {
        fun create(
            name: String,
            phone: String,
            email: String,
            address: String,
            adoption: Adoption? = null,
            care: Care? = null
        ): Applicant {
            return Applicant(
                name = name,
                phone = phone,
                email = email,
                address = address,
                adoption = adoption,
                care = care
            )
        }
    }
}
